client/comms/wsconn.go

```diff
@@ -0,0 +1,377 @@
+package comms
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/decred/dcrdex/server/comms/msgjson"
+	"github.com/gorilla/websocket"
+)
+
+const (
+	// pingWait is the maximum time in seconds to wait for a ping from
+	// the server. The client can be reconnected after this threshold since
+	// the websocket connection to the server is assumed to broken.
+	pingWait = 60 * time.Second
```

Comment:
This is the correct value, but I'm wondering how best to handle the fact that it's hard coded here. If the server were to switch to a 30 second ping, it would break the client. No need to change anything right now, though.

Reply:
That's a constant, it's actually the value set on `WsCfg` the client uses. Same goes for `WriteWait`.